subsys/bluetooth/host/mesh/models/generic_battery_srv.c

```diff
@@ -0,0 +1,89 @@
+/* Bluetooth Mesh Models */
+
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <net/buf.h>
+
+#include <bluetooth/mesh.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_MODEL)
+#include "common/log.h"
+
+static inline void u32_to_le24(u32_t value, u8_t *u24)
```

Comment:
A `buf_add_le24()' might save even more code if you foresee needing this in many places, i.e. something that takes a net_buf_simple and the host-byteorder value.

Reply:
I'll update it according to you comment - sounds reasonable.

Reply:
I did a research in another models and none of them (only generic battery) has 24bit data. btw. This operation with conversion is already implemented in sample/bluetooth/peripheral_esp. IMHO adding net_buff_add_le24() for now will generate more effort than benefit.

Reply:
There's a benefit even with two users, IMO. Also, I didn't mean an official net_buf_simple API - it can still be a static function like it is now (that's why I called it buf_add_le24 instead of net_buf_simple_add_le24)

Reply:
Ok, I've changed it.